### DISCUSSION TITLE
resize osg viewer window dynamically

### DISCUSSION THREAD

**Discussion by CapTainHbb** [openscenegraph/OpenSceneGraph#1199] [Q&A] [answered]
i want to resize osg viewer window dynamically based on an external event like pressing 'a' key on keyboard. through reading the Veiwer.cpp codes in osg core, i found this line of code for resizing window:
`getEventQueue()->windowResize(newX, newY, newWidth, newHeight);     `

so i used it in my code:

```
while (!viewer->done()) {
    viewer->frame();
    if (bConditionMet) viewer->getEventQueue()->windowResize(newX, newY, newWidth, newHeight);      
}
```
but when i call this function it seems that it is ineffective (window size and location doesn't change) . so the question is how can i do this job properly?

**Comment by robertosfield** [comment, marked as answer]
The EventQueue is for recording events sent from the windowing system, it doesn't have any actions associated with, so eventQueue->windowResize(..) just say that a window resize event has happened.

If you you want to resize a window then you should call [window->setWindowRectange(x,y,width,height)](https://github.com/openscenegraph/OpenSceneGraph/blob/master/include/osgViewer/GraphicsWindow#L60).

**Reply by CapTainHbb** [reply]
yes! that worked! thank you very much!
